src/Pelagos/Bundle/AppBundle/Command/DoiDatacitePostMigrationCommand.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Pelagos\Bundle\AppBundle\Command;
+
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\ORMException;
+
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+use Pelagos\Entity\Dataset;
+use Pelagos\Entity\DatasetSubmission;
+use Pelagos\Entity\DOI;
+use Pelagos\Entity\DIF;
+
+use Pelagos\Util\DOIutil;
+
+/**
+ * Post-migration Script for moving from EZid Library to Datacite.
+ *
+ * @see ContainerAwareCommand
+ */
+class DoiDatacitePostMigrationCommand extends ContainerAwareCommand
+{
+    /**
+     * Configures the current command.
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('dataset-doi:post-migration-script')
+            ->setDescription('Recreate DOIs for DIF approved datasets.');
+    }
+
+    /**
+     * Executes the current command.
+     *
+     * @param InputInterface  $input  An InputInterface instance.
+     * @param OutputInterface $output An OutputInterface instance.
+     *
+     * @throws \Exception When dataset not found.
+     * @throws \Exception When datasetSubmission not found.
+     *
+     * @return integer Return 0 on success, or an error code otherwise.
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $entityManager = $this->getContainer()->get('doctrine.orm.entity_manager');
+        //Recreate the DOIs which we deleted.
+        $this->recreateDoi($entityManager);
+
+        return 0;
+    }
+
+    /**
+     * Deletes DOI property from Dataset and object for DIF approved datasets.
+     *
+     * @param EntityManager $entityManager An instance of EntityManager.
+     *
+     * @throws \Exception Exception thrown when create DOI fails.
+     *
+     * @return void
+     */
+    private function recreateDoi(EntityManager $entityManager)
+    {
+        $datasets = $entityManager->getRepository(Dataset::class)->findBy(array(
+            'metadataStatus' => array(
+                DatasetSubmission::METADATA_STATUS_NONE
+            )
+        ));
+
+        foreach ($datasets as $dataset) {
+            // Creating DOI's which are DIF approved datasets.
+            if ($dataset->getIdentifiedStatus() === DIF::STATUS_APPROVED) {
+                $doi = $dataset->getDoi();
```

Comment:
Instead of doing all this, I would just publish the issue command for the consumer.

`$this->get('old_sound_rabbit_mq.doi_issue_producer')->publish($dif->getDataset()->getId(), 'issue');`

Reply:
If you look at issueDoi on the doiConsumer you can see that this is nearly the same code.

Reply:
Yeah, the pre-migration script needed changes in the consumer, so i called the Util methods directly from the script, followed the same here :P. Calling the consumer makes more sense in this script. 👍